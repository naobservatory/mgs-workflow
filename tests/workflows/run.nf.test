nextflow_pipeline {

    name "Test workflow RUN"
    script "main.nf" // Running from main so we have access to published files
    config "tests/run.config"
    tag "workflow"

    test("RUN workflow should run without errors") {
        tag "run"
        tag "main_run"
        then {
            // Should run without failures
            assert workflow.success
        }
    }

    test("RUN workflow output should match snapshot") {
        tag "run_output"
        tag "main_run_output"
        then {
            // Should run without failures
            assert workflow.success
            // Should match expected output
            def base_path = "${launchDir}/output/results/"
            def gold_files = ["bracken_reports_merged.tsv.gz", "kraken_reports_merged.tsv.gz",
                "subset_qc_adapter_stats.tsv.gz", "subset_qc_basic_stats.tsv.gz",
                "subset_qc_quality_base_stats.tsv.gz", "subset_qc_quality_sequence_stats.tsv.gz",
                "subset_qc_length_stats.tsv.gz", "virus_clade_counts.tsv.gz", "virus_hits_db.tsv.gz",
                "blast_hits_paired.tsv.gz", "virus_hits_1.fasta.gz", "virus_hits_2.fasta.gz",
                "read_counts.tsv.gz"]
            def snapshot_matches = []
            for (file in gold_files) {
                snapshot_matches += snapshot(path(base_path + file)).match("run_output")
            }
            println(snapshot_matches)
            for (snap in snapshot_matches) {
                assert snap
            }
        }
    }

}
