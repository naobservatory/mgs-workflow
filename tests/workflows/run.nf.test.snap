{
    "run_output": {
        "content": [
            "bracken_reports_merged.tsv.gz:md5,6c504fa837ef97ef2096f2569d8c6902",
            "kraken_reports_merged.tsv.gz:md5,84f070b42b948d36ae38eaee4a61982e",
            "merged_blast_filtered.tsv.gz:md5,b26a764f7b7271256c0d58a89b5517eb",
            "merged_blast_input_subset.fasta.gz:md5,f4e39b1d55362b449914a9544e8741f2",
            "read_counts.tsv.gz:md5,8dc2e3ad82f42202262a5e67a9d91e1b",
            "subset_qc_adapter_stats.tsv.gz:md5,43a90fc81f11a57e191f10176d3b7caf",
            "subset_qc_basic_stats.tsv.gz:md5,98699e1e92085c89771f0a46fa54df0d",
            "subset_qc_length_stats.tsv.gz:md5,deb6f5182ed916986640d4668459ddae",
            "subset_qc_quality_base_stats.tsv.gz:md5,a5492c9c3b986d99848b95328b32ea7b",
            "subset_qc_quality_sequence_stats.tsv.gz:md5,94c11e0dbcf2b58d685ab8701c04cf9b",
            "virus_hits_filtered.tsv.gz:md5,0e57eaaf38f8605340d2295063c8c7e9"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-11T19:14:28.184957513"
    }
}